clang/lib/Driver/Driver.cpp

```diff
@@ -3678,6 +3678,13 @@ bool Driver::checkForSYCLDefaultDevice(Compilation &C,
 // static offload libraries.
 bool Driver::checkForOffloadStaticLib(Compilation &C,
                                       DerivedArgList &Args) const {
+  // Make -fintelfpga flag imply -fsycl by default.
+  const llvm::opt::OptTable &Opts = getOpts();
+  Arg *SYCLFpgaArg = C.getInputArgs().getLastArg(options::OPT_fintelfpga);
+  if (SYCLFpgaArg &&
+      !Args.hasFlag(options::OPT_fsycl, options::OPT_fno_sycl, false))
+    Args.AddFlagArg(0, Opts.getOption(options::OPT_fsycl));
+
```

Comment:
Instead of adding the flag here, what do you think about adding in `TranslateInputArgs` instead?  Seems to be a better fit there.

Reply:
Updated.